### PR TITLE
Remove EP-Compare

### LINKED ISSUE #7324: EP-Compare is outdated for modern macs
Issue overview
--------------

The EP-Compare is a outdated to run on modern mac. Fixing the permissions doesn't help really.

```
chmod +x /Applications/EnergyPlus-9-1-0/PostProcess/EP-Compare/EP-Compare.app/Contents/MacOS/EP-Compare
```

If I chmod +x I still can't launch it by clicking on the App (which shows a O/ sign).: "You can't open the application "EP-Compare" because it may be damaged or incomplete"

If I launch it from the terminal, it complains but eventually launches anyway. I did briefly also get a messagebox saying the app needed an update from the developpers.

```
./EP-Compare
/BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.60.1/Common/ChunkCompression.cpp:49: Error: unsupported compressor 8
/BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.60.1/Libraries/CompressData/CompressData.c:353: Error: Unknown compression scheme encountered for file '/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/Exceptions.plist'
/BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.60.1/Common/ChunkCompression.cpp:49: Error: unsupported compressor 8
/BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.60.1/Libraries/CompressData/CompressData.c:353: Error: Unknown compression scheme encountered for file '/System/Library/CoreServices/CoreTypes.bundle/Contents/Library/AppExceptions.bundle/Exceptions.plist'
```

I see only binary files, not development files in https://github.com/NREL/EnergyPlus/tree/develop/bin/EP-Compare/Run-Mac/EP-Compare.app/Contents. **No clue where the source is.**

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by JasonGlazer**
Pull request overview
---------------------

Removes EP-Compare program 

The EP-Compare program did not appear to have any users since it showed an issue when being used that was never reported by any user.

Removes the binaries and updates the documentation and leaves a readme and a brief note in the documentation only.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by shorowit** [comment]
Fixes https://github.com/NREL/EnergyPlus/issues/7324? 😄

**Comment by JasonGlazer** [comment]
> Fixes #7324? 😄

I suppose, if you really stretch the concept of a "fix" :-)

File: cmake/CPack.STGZ_Header.sh.in [hunk 1]
```diff
@@ -141,8 +140,6 @@ if [ ! "$link_directory" = "n" ]; then
 		rm -f "${link_directory}/energyplus"
 		rm -f "${link_directory}/Energy+.idd"
 		rm -f "${link_directory}/Energy+.schema.epJSON"
-		rm -f "${link_directory}/EP-Compare"
-		rm -f "${link_directory}/EPMacro"
```
Comment (jmarrec) [inline cmake/CPack.STGZ_Header.sh.in:145]: Did you remove that EPMacro line by mistake?

**Review by jmarrec — CHANGES_REQUESTED** [review]
There are still a couple of locations that need to be adjusted.

```
scripts/dev/verify_signature.py
76:    "PostProcess/EP-Compare/EP-Compare.app",


src/EP-Launch/epl-ui.frm
6078:utilProg(10).name = "EP-Compare"
6091:utilProg(10).applicationFile = "PostProcess\EP-Compare\EP-Compare.exe"

doc/getting-started/src/getting-started-with-energyplus.tex
37:|   +-- EPCompare              A graphical tool for comparing two EnergyPlus output sets
```

File: cmake/CPack.STGZ_Header.sh.in (hunk 1 above)
Comment (JasonGlazer) [reply cmake/CPack.STGZ_Header.sh.in:145]: Thanks @jmarrec, the EPMacro line was a mistake and I will make the changes to verify_signature.py and getting started. 

The change to EP-Launch, I think I will skip for now. It fails gracefully, which I think is better than just making it disappear from the list of choices. But the next time I work on EP-Launch, I will pick it up for a future release. I added it to this [issue for the future](https://github.com/NREL/EnergyPlus/issues/10576)

Comment (JasonGlazer) [reply cmake/CPack.STGZ_Header.sh.in:145]: @jmarrec in verify_signature.py, shouldn't the list of the bundled apps include others like IDFVersionUpdater, ExpandObjects and EPMacro and other utilities that are available on MacOS? Or is it only for apps that had not included the source code in the repo?

**Comment by JasonGlazer** [comment]
I just posted the source and zip of the binaries in an archive repo: 

https://github.com/JasonGlazer/ep-compare

File: cmake/CPack.STGZ_Header.sh.in (hunk 1 above)
Comment (jmarrec) [reply cmake/CPack.STGZ_Header.sh.in:145]: IDFVersionUpdater is in there. That BUNDLED_APPS is just so I don't recheck something I actually codesigned in the repo directly.

ExpandObjects is a built one (and a fortran exe, not a bundled app) so it'll be checked by default.

https://github.com/NREL/EnergyPlus/blob/22e9501303d86afcec015a1247d6ae32f4e7aeee/scripts/dev/verify_signature.py#L73-L77

**Review by jmarrec — APPROVED** [review]

**Review by Myoldmopar — APPROVED** [review]
Love this.  Great job cleaning it all out and putting it in a separate archive repo.  I'll make sure to note this in the readme and deprecation files that go into the release packages.

GitHub is of course showing that you removed about 235 lines of code with this PR, it would be kinda nice if it actually said how many MB you were also removing with all the binary removals.

**Comment by Myoldmopar** [comment]
It's like 45 MB removed, nice!!  OK, this is great.  Merging this in.  Thanks @JasonGlazer and @jmarrec.  Obviously we'll make sure things are happy when I start making release candidates and deal with issues at that point.

**Pull request merged**